src/System.Private.CoreLib/shared/System/ReadOnlySpan.Fast.cs

```diff
@@ -154,6 +154,37 @@ internal ReadOnlySpan(ref T ptr, int length)
 #endif
         }
 
+        public ref readonly T this[Index index]
+        {
+#if PROJECTN
+            [BoundsChecking]
+            get
+            {
+                return ref Unsafe.Add(ref _pointer.Value, index.FromEnd ? _length - index.Value : index.Value);
+            }
+#else
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
```

Comment:
Note that the regular indexer is treated as JIT intrinsics. This new indexer won't be treated as JIT intrinsic. So this will hurt some case and improve others. I think we should start with the simple code you had here and only add the more complex one once we have performance data that it is an actual improvement.